ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/AddDiscriminatedUnionAllClauses.fs

```diff
@@ -0,0 +1,54 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+open FSharp.Compiler.SourceCodeServices
+
+type AddDiscriminatedUnionAllClauses(warning: MatchIncompleteWarning) =
+    inherit FSharpQuickFixBase()
+
+    override x.Text = "Autogenerate all discriminated union cases"
+
+    override x.IsAvailable _ =
+        let refExpr = warning.Expr.Expression.As<IReferenceExpr>()
```

Comment:
It should work in much more cases than a simple identifier/value (e.g. a function return, an indexer access, union case creation, etc). Try getting expression type and checking if it's a union instead.